frontend/test/resolution/testResolve.cpp

```diff
@@ -830,6 +830,242 @@ static void test19() {
+    const Variable* x = m->stmt(2)->toVariable();
+    assert(x);
+    const AstNode* rhs = x->initExpression();
+    assert(rhs);
+    const FnCall* fnCall = rhs->toFnCall();
+    assert(fnCall);
+
+    // Get called foo
+    const ResolutionResultByPostorderID& rr = resolveModule(context, m->id());
+    const ResolvedExpression& re = rr.byAst(fnCall);
+
+    // Check foo call is ambiguous
+    assert(re.mostSpecific().isAmbiguous());
+
+    // Check variable couldn't resolve
+    auto rrx = rr.byAst(x);
+    auto xt = rrx.type().type();
+    assert(xt);
+    assert(xt->isErroneousType());
+
+    assert(guard.numErrors() == 1);
+    assert(guard.error(0)->message() ==
+           "Cannot resolve call to 'foo': ambiguity");
+    guard.realizeErrors();
+
+    context->collectGarbage();
+  }
+
+  // Test resolving a last resort that's the only actual option.
+  {
+    printf("part 3\n");
+    context->advanceToNextRevision(true);
+
+    auto path = UniqueString::get(context, "input.chpl");
+    std::string contents = R""""(
+        module M {
+          proc foo(x: string) {
+            return x;
+          }
+          pragma "last resort"
+          proc foo(y: int) {
+            return y;
+          }
+          var x = foo(4);
+        }
+                          )"""";
+
+    setFileText(context, path, contents);
+
+    const ModuleVec& vec = parseToplevel(context, path);
+    assert(vec.size() == 1);
+    const Module* m = vec[0]->toModule();
+    assert(m);
+    assert(m->numStmts() == 3);
+
+    // variable initialized with foo call
+    const Variable* x = m->stmt(2)->toVariable();
+    assert(x);
+    const AstNode* rhs = x->initExpression();
+    assert(rhs);
+    const FnCall* fnCall = rhs->toFnCall();
+    assert(fnCall);
+
+    // Get called foo
+    const ResolutionResultByPostorderID& rr = resolveModule(context, m->id());
+    const ResolvedExpression& re = rr.byAst(fnCall);
+    auto c = re.mostSpecific().only();
+    assert(c);
+
+    // Check we called the correct foo.
+    assert(c.fn()->untyped()->name() == "foo");
+    assert(c.fn()->numFormals() == 1);
+    assert(c.fn()->formalName(0) == "y");
+    assert(c.fn()->formalType(0).type());
+    assert(c.fn()->formalType(0).type()->isIntType());
+
+    context->collectGarbage();
+  }
+
+  // Test resolving between forwarded-to methods where just one is last resort.
+  {
+    printf("part 4\n");
+    context->advanceToNextRevision(true);
+
+    auto path = UniqueString::get(context, "input.chpl");
+    std::string contents = R""""(
+        module M {
+          record MyImpl {
+            proc foo(x: int) {
+              return x;
+            }
+            pragma "last resort"
+            proc foo(y: int) {
+              return y;
+            }
+          }
+          record MyForward {
+            forwarding var impl: MyImpl;
+          }
+          var mf: MyForward;
+          var x = mf.foo(4);
+        }
+                          )"""";
```

Comment:
I'd be interested to see the following case:

```Chapel
       module M {
          record MyImpl {
            proc foo(x: int) {
              return x;
            }
          }
          record MyForward {
            forwarding var impl: MyImpl;

            pragma "last resort"
            proc foo(y: int) {
              return y;
            }
          }
          var mf: MyForward;
          var x = mf.foo(4);
        }
```

(i.e., the last-resort candidate is in a different scope, provided by the forwarder. I'd envision this to be a useful feature, because you can say "forward this method, but if my underlying type doesn't have it, fall back to my own implementation". 

E.g. a container for a numeric type might try to forward the `pow` method, but if one isn't defined, it can try exponentiate itself using repeated multiplications.